core/action/config.go

```diff
@@ -1,10 +1,32 @@
 package action
 
-import "encoding/json"
+import (
+	"encoding/json"
+	"github.com/TIBCOSoftware/flogo-lib/core/data"
+)
 
 // Config is the configuration for the Action
 type Config struct {
-	Id   string          `json:"id"`
-	Ref  string          `json:"ref"`
-	Data json.RawMessage `json:"data"`
+	Id       string          `json:"id"`
+	Ref      string          `json:"ref"`
+	Data     json.RawMessage `json:"data"`
+	Metadata *ConfigMetadata `json:"metadata"`
```

Comment:
Its not really an input/output, we have that else where.  Its the definition of the Input/Output, I don't want to confuse it... maybe another word other than metadata?  I chose metadata because we use it elsewhere to imply the definition of things.